quil-py/quil/program/__init__.pyi

```diff
@@ -1,3 +1,70 @@
+"""
+The `quil.program` module contains classes for constructing and representing a Quil program.
+
+# Examples
+
+## Source Mapping for Calibration Expansion
+
+```py
+    import inspect
+
+    program_text = inspect.cleandoc(
+        \"\"\"
+        DEFCAL X 0:
+            Y 0
+
+        DEFCAL Y 0:
+            Z 0
+
+        X 0 # This instruction is index 0
+        Y 0 # This instruction is index 1
+        \"\"\"
+    )
+
+    # First, we parse the program and expand its calibrations
+    program = Program.parse(program_text)
+    expansion = program.expand_calibrations_with_source_map()
+    source_map = expansion.source_map()
+
+    # This is what we expect the expanded program to be. X and Y have been replaced by Z
+    expected_program_text = inspect.cleandoc(
+        \"\"\"
+        DEFCAL X 0:
+            Y 0
+
+        DEFCAL Y 0:
+            Z 0
+
+        Z 0 # This instruction is index 0
+        Z 0 # This instruction is index 1
+        \"\"\"
+    )
```

Comment:
I think this ought to have an assert.  That raises questions of the precise whitespace, I know.  Could we parse them both to Quil and assert the parsed programs are equal?

Reply:
the test did, I left it out here for readability but have restored it now